.github/workflows/clang_format.yml

```diff
@@ -0,0 +1,34 @@
+name: Clang Format
+
+on:
+  workflow_dispatch:
+  push:
+    branches: ["main"]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ["main"]
+
+jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Clang Format
+        run: |
+          # TODO: Figure out how always to install the latest version of clang-format
+          # This LLVM script will add the relevant LLVM PPA: https://apt.llvm.org/
+          wget https://apt.llvm.org/llvm.sh -O /tmp/llvm.sh
+          chmod +x /tmp/llvm.sh
+          sudo /tmp/llvm.sh 18
+          sudo apt-get install -y clang-format-18 clang-tidy-18
+          rm /tmp/llvm.sh
+
+      - name: Clang format version
+        run: clang-format-18 --version
+
+      - name: Run Clang Format
+        run: |
+          find . -name "*.h" -o -name "*.c" -o -name "*.cpp" -name "*.mm" | xargs clang-format-18 --style=file --fallback-style=none --Werror --dry-run
+          find cpp -name "*.m" | xargs clang-format-18 --style=file --fallback-style=none --Werror --dry-run
```

Comment:
How this work, does clang-format return a different status code if there are changes?

Reply:
`--Werror` with `--dry-run` causes it to return an error code. Maybe we can find a nicer way to call it

Reply:
See https://github.com/zeroc-ice/ice/actions/runs/8205903763/job/22443823637 for example